src/webpack/plugin/SitePlugin.js

```diff
@@ -0,0 +1,287 @@
+/* eslint-disable no-param-reassign */
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const path = require('path');
+const { DefinePlugin } = require('webpack');
+const fs = require('fs');
+
+const DirectorySwitcherPlugin = require('./resolve/DirectorySwitcherPlugin');
+const LocalPackageAliasPlugin = require('./resolve/LocalPackageAliasPlugin');
+const { babelLoader, mdxOptions, getMdxLoader } = require('./siteLoaderUtils');
+const getNewRelicJS = require('../new-relic/getNewRelicJS');
+
+// Singletons
+let oneTimeSetupComplete = false;
+const siteRegistry = {};
+const processPath = process.cwd();
+const isLernaMonoRepo = fs.existsSync(path.join(processPath, 'lerna.json'));
+
+/**
+ * Updates the webpack options with defaults that terra-dev-site requires.
+ */
+class SitePlugin {
+  constructor({
+    entry,
+    config,
+    applyDefaults,
+    contentDirectory,
+  } = {}) {
+    // Apply defaults to the config.
+    this.siteConfig = applyDefaults(config);
+    this.contentDirectory = contentDirectory;
+    const { pathPrefix, titleConfig } = this.siteConfig;
+    this.entry = entry;
+
+    if (pathPrefix) {
+      this.entryKey = `${pathPrefix}/index`;
+      this.resourceQuery = `?${pathPrefix}-terra-entry`;
+      this.htmlFileName = `${pathPrefix}/index.html`;
+      this.url = `/${pathPrefix}/`;
+    } else {
+      this.entryKey = 'index';
+      this.resourceQuery = '?terra-entry';
+      this.htmlFileName = 'index.html';
+      this.url = '/';
+    }
+
+    if (siteRegistry[pathPrefix]) {
+      throw Error('The PathPrefix must be unique per TerraDevSite Plugin');
+    }
+    // Register each application instance with the siteRegistry
+    siteRegistry[pathPrefix] = {
+      path: pathPrefix,
+      url: this.url,
+      title: titleConfig.title,
+      entry: this.entryKey,
+    };
+  }
+
+  static applyOneTimeSetup({
+    compiler,
+    sourceFolder,
+    distributionFolder,
+    basename,
+  }) {
+    if (oneTimeSetupComplete) {
+      return;
+    }
+    oneTimeSetupComplete = true;
+
+    compiler.options.entry = {
+      ...compiler.options.entry,
+      rewriteHistory: '@cerner/terra-dev-site/lib/browser-router-redirect/rewriteHistory',
+      redirect: '@cerner/terra-dev-site/lib/browser-router-redirect/redirect',
+    };
+
+    // MODULE
+    const mdxLoader = getMdxLoader(compiler.options.output.publicPath);
```

Comment:
Why break out getMdxLoader and babelLoader, but none of the other configs used here?

Reply:
because the babel and mdx loaders are reused throughout the other loaders.